tests/inst/src/destructive.rs

```diff
@@ -0,0 +1,580 @@
+            .context("Failed to parse reboot mark")?
+    } else {
+        RebootMark {
+            ..Default::default()
+        }
+    };
+    // Drop the &mut
+    let commitstates = commitstates;
+
+    let mut rt = tokio::runtime::Runtime::new()?;
+    let cycle_time_ms = (tdata.cycle_time.as_secs_f64() * 1000f64 * FORCE_REBOOT_AFTER_MUL) as u64;
+    // Explicitly keep track of
+    let mut live_strategy: Option<InterruptStrategy> = None;
+    let mut retries = 0;
+    // This loop is for the non-rebooting strategies - we might use kill -9
+    // or not interrupt at all.  But if we choose a reboot strategy
+    // then we'll exit implicitly via the reboot, and reenter the function
+    // above.
+    loop {
+        if let Some(last_strategy) = live_strategy.take() {
+            mark.iter += 1;
+            retries = 0;
+            let res = validate_live_interrupted_upgrade(&commitstates)?;
+            mark.get_results_map(&last_strategy)
+                .entry(res)
+                .and_modify(|result_e| {
+                    *result_e += 1;
+                })
+                .or_insert(1);
+        }
+        if mark.iter == ITERATIONS {
+            // TODO also add ostree admin fsck to check the deployment directories
+            sh_execute!(
+                "echo Performing final validation...
+                ostree fsck"
+            )?;
+            return Ok(());
+        }
+        let mut rng = rand::thread_rng();
+        let strategy: InterruptStrategy = if rand::thread_rng()
+            .gen_ratio(FORCE_INTERRUPT_PERCENTAGE, 100)
+        {
+            InterruptStrategy::Force(force_strategies.choose(&mut rng).expect("strategy").clone())
+        } else {
+            InterruptStrategy::Polite(
+                polite_strategies
+                    .choose(&mut rng)
+                    .expect("strategy")
+                    .clone(),
+            )
+        };
+        println!("Using interrupt strategy: {:?}", strategy);
+        // Initiate a forced reboot usually the upgrade would
+        // complete, but also ~30% of the time after.
+        let sleeptime = time::Duration::from_millis(rng.gen_range(0, cycle_time_ms));
+        println!(
+            "force-reboot-time={:?} cycle={:?} status:{:?}",
+            sleeptime, tdata.cycle_time, &mark
+        );
+        sh_execute!(
+            "
+            rpm-ostree status -v
+            systemctl stop rpm-ostreed
+            ostree reset testrepo:{testref} {booted_commit}
+            rpm-ostree cleanup -pbrm
+            ",
+            testref = TESTREF,
+            booted_commit = booted_commit
+        )
+        .context("Failed pre-upgrade cleanup")?;
+        let res: Result<bool> = rt.block_on(async move { run_upgrade_or_timeout(sleeptime).await });
+        let res = res.context("Failed during upgrade")?;
+        if res {
+            println!(
+                "Failed to interrupt upgrade, attempt {}/{}",
```

Comment:
We could lower the timeout by e.g. 10% in this case to increase the odds for the next time (maybe after we get to 5 retries or something).

Reply:
I thought about doing things like this - an issue I've seen is that when I start up 4 instances of this test in parallel, there's heavy CPU usage in the VMs where they're all doing the initial setup, and that inflates the timing for the test upgrade.

The better fix I think would be something like an `env OSTREE_PAUSE_POINT=pre-deploy,post-deploy,pre-cleanup` where the harness can control when the process continues.  (Or maybe implement this with scripting gdb or so)